Sources/SPTDataLoader/SPTDataLoaderRequestTaskHandler.m

```diff
@@ -235,7 +235,8 @@ - (void)checkRetryLimiterAndExecute
         }
         return;
     }
-    
+
+    self.receivedData = nil;
```

Comment:
It seems like this problem might not be happening because `self.response` is always re-assigned in `receiveResponse:`, which according to my tests should be called when all kind of responses are received, failing or succeeding. 

But I think that it's worth assigning the nil anyway so the state is updated consistently. Will do!

Reply:
Hmm, doing so would require to change the `response` property to `nullable` which it currently is not. However, adding nullability to it creates number of `Implicit conversion from nullable pointer to non-nullable pointer type` errors, fixing which is slightly non-ergonomic in Objective-C. I'm not sure if it's worth it considering that `self.response = nil` would be rather a precaution and not a necessity.

Reply:
OK, it's something we can revisit later if need be. It caught my eye because it's unclear whether `-[SPTDataLoaderRequestTaskHandler receiveResponse:]` is guaranteed to be called before `-[SPTDataLoaderRequestTaskHandler completeWithError:]` — if it's not, then the response could contain incorrect data on a failed retry attempt.